avogadro/rendering/curvegeometry.cpp

```diff
@@ -69,39 +79,71 @@ void CurveGeometry::update(int index)
 
   const size_t qttySegments = lineResolution * qttyPoints;
   Vector3f previous;
-  const size_t skip = 1;
   std::vector<Eigen::Affine3f> points;
   size_t top = qttyPoints <= 4 ? 0 : line->points.size() - 4;
-  for (size_t i = skip; i < top; ++i) {
+  auto it = line->points.begin();
+  for (size_t i = SKIPED; i < top; ++i) {
     for (size_t j = 0; j < lineResolution; ++j) {
-      auto p = computeCurvePoint((i * lineResolution + j) / float(qttySegments),
-                                 line->points);
-      if (i > skip) {
+      float t = (i * lineResolution + j) / float(qttySegments);
+      auto p = computeCurvePoint(t, line->points);
+      if (i > SKIPED) {
         Eigen::Matrix3f m;
         m.col(1) = (p - previous).normalized();
         m.col(0) = m.col(1).unitOrthogonal() * -1.0f;
+        if (i > SKIPED + 1) {
+          const auto& previousAngle = points.back().linear().col(0);
+          float angle = previousAngle.dot(m.col(0));
+          // avoid nans
+          if (std::isnan(angle)) {
+            m.col(0) = previousAngle;
+            break;
+            // if angle > 90 flip it
+          } else if (angle <= 0.0f) {
+            m.col(0) *= -1.0f;
+          }
+          angle = previousAngle.dot(m.col(0));
+          float degrees =
+            (std::acos(angle / (m.col(0).norm() * previousAngle.norm())) *
+             180.0f / 3.1415f);
```

Comment:
see `core/avogadrocore.h` especially `PI_F` already as a macro.
(I'd be happy to add `DEG_TO_RAD_F` and `RAD_TO_DEG_F` as well.)